release/downloadIstioCandidate.sh

```diff
@@ -42,35 +42,57 @@ if [ "x${ISTIO_VERSION}" = "x" ] ; then
 		  grep -v -E "(alpha|beta|rc)\.[0-9]$" | sort -t"." -k 1,1 -k 2,2 -k 3,3 -k 4,4 | tail -n 1)
 fi
 
+if [ "x${ISTIO_ARCH}" = "x" ]; then
+  ISTIO_ARCH=$(arch)
+fi
+
+#for now this supports amd64, which has arch output of x86_64. Thhe arch command will let us switch based on host
+ISTIO_ARCH="amd64"
+
 if [ "x${ISTIO_VERSION}" = "x" ] ; then
   printf "Unable to get latest Istio version. Set ISTIO_VERSION env var and re-run. For example: export ISTIO_VERSION=1.0.4"
   exit;
 fi
 
 NAME="istio-$ISTIO_VERSION"
 URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-${OSEXT}.tar.gz"
+ARCH_URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-${OSEXT}-${ISTIO_ARCH}.tar.gz"
+
 printf "Downloading %s from %s ..." "$NAME" "$URL"
-if ! curl -L "$URL" | tar xz
+if ! curl -fsLO "$URL"
 then
-  printf "\n\n"
-  printf "Unable to download Istio %s at this moment!\n" "$ISTIO_VERSION"
-  printf "Please verify the version you are trying to download.\n\n"
+  printf "Failed.\n\n Trying with ISTIO_ARCH. Downloading %s from %s ...\n" "$NAME" "$ARCH_URL"
+  if ! curl -fsLO "$ARCH_URL"
+  then
+    printf "\n\n"
+    printf "Unable to download Istio %s at this moment!\n" "$ISTIO_VERSION"
+    printf "Please verify the version you are trying to download.\n\n"
+    exit
+  else
+    filename="istio-${ISTIO_VERSION}-${OSEXT}-${ISTIO_ARCH}.tar.gz"
```

Comment:
This won't work for the files without the arch like: istio-1.6.0-alpha.2-osx.tar.gz

Reply:
Set filename without the ISTIO_ARCH before the first curl, then update with ISTIO_ARCH inside the if?

Reply:
If I understand correctly, there's nothing to address here. Right?

Reply:
Correct